File: bitMasking/triplet.go
package bit

import "fmt"

//FindTriplet : Given an array we need to find triplet such that 0<i<j<=k
func FindTriplet(arr []int) {
	/*
			And they should full fil this condition also
			 X = arr[i] ^ arr[i+1] ^ ... ^ arr[j - 1]
		    Y = arr[j] ^ arr[j+1] ^ ... ^ arr[k]
	*/

	var count int

	for i := 0; i < len(arr)-1; i++ {
		fmt.Println("The value of i is ", i)
		current := arr[i]
		for j := i + 1; j < len(arr); j++ {
			current = current ^ arr[j]
			if current == 0 {
				count += j - i
			}
		}

	}
	fmt.Println(count)

}

File: arrays/triplet.py
#given an array we need to return true or false such that if there exist a tiple such that a+b+c is equal to the X
def findTriplet(arr,N,X):
    start=0
    arr.sort()
    for i in range(len(arr)-2):
        start=i+1
        end=len(arr)-1
        while start<end:
            current=arr[i]+arr[start]+arr[end]
            if current==X:
                return True
            elif current<X:
                start+=1
            else:
                end-=1
    return False


arr=[1,2,4,3,6]
ans=findTriplet(arr,5,10)
print(ans)